lib/tokenize.cpp

```diff
@@ -827,6 +827,10 @@ namespace {
                 }
             }
 
+            // don't add class|struct|union in inheritance list
+            if (Token::Match(tok->previous(), "public|private|protected") && Token::Match(mRangeType.first, "class|struct|union"))
+                mRangeType.first = mRangeType.first->next();
```

Comment:
It does not feel proper to change `mRangeType` in `replace` this affects all subsequent replacements also. Maybe replace should be const but then the mIsUsed must be mutable..

Reply:
We could probably skip `class|struct|union|enum` in C++ code altogether, since its redundant and can cause other issues too: https://trac.cppcheck.net/ticket/11030
That problem is in `simplifyTypedefCpp()` though.

Reply:
But `replace()` should not be const since it has side effects.

Reply:
> We could probably skip class|struct|union|enum in C++ code altogether, since its redundant and can cause other issues too:

yes.

> But replace() should not be const since it has side effects.

I agree.

Reply:
Is there anything left to do here?

Reply:
> But replace() should not be const since it has side effects.

if there are only side effects to the non-const parameter that is passed to it I feel it's logical to make it const.. but we do have the mIsUsed member also..

Reply:
I have the feeling it can be an advantage in further processing to keep a redundant class|struct in case we don't see the class/struct declaration.. if there is redundant class|struct keyword token cppcheck will still know that it's a type and that it's some kind of record. 

I can't point out a concrete example though.

I am not sure.. I have a vague feeling it could be better to make a copy of mRangeType and modify that instead.

Reply:
here is one more problematic example code:

```
struct B {};
typedef const struct B B1;
namespace N {
    struct D : public B1 {};
}
```